### PR TITLE
Update build scripts

### PR DESCRIPTION
**Pull request by ethomson**
Update the build scripts to produce a `linux-x64` package.

The build scripts will now propagate failures, so CI runs will fail if artifacts were not produced (instead of giving the appearance of success, and ultimately producing nupkgs that lacked the artifacts).  The linux and rhel images now install a newer version of cmake, required for building libgit2.

### CODE REVIEW

File: Dockerfile.linux-x64 [hunk 1]
```diff
@@ -2,6 +2,8 @@ FROM ubuntu:14.04
 WORKDIR /nativebinaries
 COPY . /nativebinaries/
 
-RUN apt update && apt -y install cmake libssl-dev pkg-config
+RUN apt update && apt -y install libssl-dev pkg-config curl make gcc build-essential
```
Comment (bording) [inline Dockerfile.linux-x64]: Out of curiosity, why did you remove the old cmake from being installed here, but didn't alter it in the same way for the `rhel-x64` file?

Comment (ethomson) [reply Dockerfile.linux-x64]: Oversight.
